app/views/course/assessment/assessments/index.json.jbuilder

```diff
@@ -1,28 +1,130 @@
+  json.achievements achievements_enabled
+  json.bonusAttributes show_bonus_attributes?
+  json.endTimes show_end_at?
+  json.canCreateAssessments can?(:create, Course::Assessment.new(tab: @tab))
+
+  json.category do
+    json.id @category.id
+    json.title @category.title
+    json.tabs @category.tabs.each do |tab|
+      json.id tab.id
+      json.title tab.title
+    end
+  end
+
+  json.tabId @tab.id
+end
+
+json.assessments @assessments do |assessment|
   json.id assessment.id
   json.title assessment.title
-  json.description assessment.description
-  json.published assessment.published
-  json.baseExp assessment.base_exp
-  json.timeBonusExp assessment.time_bonus_exp
-  json.hasPersonalTimes assessment.has_personal_times
-  json.affectsPersonalTimes assessment.affects_personal_times
-  json.tabId assessment.tab_id
-  json.autograded assessment.autograded
-  json.sessionPassword assessment.session_password
-  json.tabbedView assessment.tabbed_view
-  json.skippable assessment.skippable
-  json.delayedGradePublication assessment.delayed_grade_publication
-  json.showPrivate assessment.show_private
-  json.showEvaluation assessment.show_evaluation
-  json.viewPassword assessment.view_password
-  json.usePublic assessment.use_public
-  json.usePrivate assessment.use_private
-  json.useEvaluation assessment.use_evaluation
-  json.allowPartialSubmission assessment.allow_partial_submission
-  json.randomization assessment.randomization
-  json.showMcqAnswer assessment.show_mcq_answer
-  json.showMcqMrqSolution assessment.show_mcq_mrq_solution
-  json.blockStudentViewingAfterSubmitted assessment.block_student_viewing_after_submitted
+
+  json.passwordProtected !assessment.view_password.nil?
+  json.published assessment.published?
+  json.autograded assessment.autograded?
+  json.hasPersonalTimes current_course.show_personalized_timeline_features && assessment.has_personal_times?
+  json.affectsPersonalTimes current_course.show_personalized_timeline_features && assessment.affects_personal_times?
+  json.url course_assessment_path(current_course, assessment)
+
+  assessment_with_loaded_timeline = @items_hash[assessment.id].actable
+  can_attempt_assessment = can?(:attempt, assessment_with_loaded_timeline)
+  json.canAttempt can_attempt_assessment
+
+  submissions = submissions_hash[assessment.id]
+  attempting_submission = submissions.find(&:attempting?)
+  submitted_submission = submissions.find { |submission| !submission.attempting? }
+
+  action_url = nil
+  if !current_course_user || !can_attempt_assessment
+    status = 'unavailable'
+  elsif cannot?(:access, assessment) && can_attempt_assessment
+    status = 'locked'
+    action_url = course_assessment_path(current_course, assessment)
+  elsif !attempting_submission.nil?
+    status = 'attempting'
+    action_url = edit_course_assessment_submission_path(current_course, assessment, attempting_submission)
+  elsif !submitted_submission.nil?
```

Comment:
Isn't `present?` a negation of `blank?`, not a negation of `nil?`. I think I used negated `nil?` because I am explicitly checking _if `*_submission` is **not `nil`**_.

Of course, `blank?` and `nil?` differs when it comes to strings. One could always argue that `present?` will work if `*_submission` will never be a string.

Reply:
Over here we know that it is of active record type and .present? will work. But, the point is we want to avoid double negation, so perhaps u could just use `submitted_submission` only instead if you dont want to use `present?`

Reply:
Then in that case, let us use `present?`.